portal-api/src/v1/users/sign-in-link.service.js

```diff
@@ -0,0 +1,69 @@
+const { ObjectId } = require('mongodb');
+const crypto = require('node:crypto');
+const sendEmail = require('../email');
+const { EMAIL_TEMPLATE_IDS, SIGN_IN_LINK_EXPIRY_MINUTES } = require('../../constants');
+const db = require('../../drivers/db-client');
+
+module.exports.createAndEmailSignInLink = async (user) => {
+  const {
+    _id: userId,
+    email: userEmail,
+    firstname: userFirstName,
+    surname: userLastName,
+  } = user;
+
+  const signInCode = await createSignInCode();
+  await saveSignInCodeHashAndSalt({ userId, signInCode });
+  return sendSignInLinkEmail({
+    userEmail,
+    userFirstName,
+    userLastName,
+    // TODO DTFS2-6750: introduce an environment variable to replace localhost
+    signInLink: `http://localhost/login/sign-in-link?t=${signInCode}`,
+  });
+};
+
+async function createSignInCode() {
```

Comment:
This sounds like a `crypto/utils.js` function to me -- what do you think?

Reply:
I'll see what I can come up with in a refactor 👍